javacutil/src/main/java/org/checkerframework/javacutil/ElementUtils.java

```diff
@@ -716,8 +716,21 @@ public static boolean matchesElement(
                         .asType()
                         .toString()
                         .equals(parameters[i].getName())) {
-
-                    return false;
+                    // might be a primitive array.
+                    String type = method.getParameters().get(i).asType().toString();
+                    if (type.contains("[]")) {
+                        int dimension = (type.lastIndexOf("]") - type.indexOf("[") + 1) / 2;
```

Comment:
Please don't do string manipulation.  It is inefficient and error-prone.
Instead, work from `method.getParameters().get(i).asType()`.